pkg/cluster/gossip.go

```diff
@@ -53,15 +54,18 @@ type GossipConfig struct {
 	// AdvertiseAddr is unset.
 	AdvertiseInterfaces flagext.StringSlice
 
-	// List of one or more host:port peer addresses to connect to. Mutually
-	// exclusive with DiscoverPeers.
+	// List of one or more host:port peer addresses to connect to.
 	//
 	// If an agent connects to no peers, it will form a one-node cluster until a
 	// peer connects to it explicitly.
 	JoinPeers flagext.StringSlice
 
+	// Comma-separated list of hosts, DNS records, or host:port peer addresses
+	// to look up. Mutually exclusive with DiscoverPeers.
+	JoinAddr string
```

Comment:
This feels off to me; we now have JoinAddr and JoinPeers, which do the same thing except JoinPeers is effectively ignored. 

We shouldn't be introducing a new field here IMO.

Reply:
The reason for this is that JoinAddr may be a DNS record that will be expanded to multiple addresses. If we use the same field, we'll lose that information during the refresh, and it will only be applicable to the DiscoverPeers.

Then JoinAddr/DiscoverPeers are the way to rediscover and JoinPeers is the actual list of peers (with expanded DNS records, go-discover addresses, the default port added etc).

Reply:
Maybe the separation of concerns is a good idea to generate the list of `<host>:<port>` addresses to join in a different way.

Reply:
Can we change the semantics so that JoinPeers is expanded if it contains a DNS record?

Reply:
> Can we change the semantics so that JoinPeers is expanded if it contains a DNS record?

Yes, but still the result has to be stored separately so that it has the chance to be expanded on the next tick as well.

Reply:
IMO it definitely feels like we should move some of that state storage to the clustering instance instead of the config.